### ISSUE TITLE
Wake Boron at predetermined time intervals

### ISSUE THREAD

**Issue by SUPScientist** [TinyCamML/Boron#18]
@BentleySettin emailed the following and I am moving it to an issue:
"I do have a question that I will try to successfully process via email. So the Boron will be hooked up to a RTC off the cloud. The RTC will help log data as a value and time we will understand and can use. As of now though I do not need the RTC because I should be able to time when the Boron wants data from the OpenMV. Therefore, I am trying to create code that puts the Boron in sleep mode for X amount of time, then will wake up, wake up the OpenMV and ask for data from the OpenMV. Both will then return to sleeping or low energy mode. And as discussed in the zoom last Tuesday the X amount of time will be 1 minute."

**Comment by SUPScientist** [comment]
Re: "real time": I think the terminology is confusing the issue. Let's think about it from a functionality-centric view (rather than what hardware we have). We want to know the actual time and we want to use that to sleep/wake at pre-determined and flexible intervals. 

Now in terms of hardware, that could happen using a peripheral RTC (such as the one on the Adalogger Featherwing or a bunch of other options) or we could just use the Boron's ability to ask the cloud what time it is. Let's use the latter! 

In other words, no need for a peripheral RTC. Based on another issue (to be created), we might not need peripheral datalogging either as we will investigate using the OpenMV's storage for datalogging. 

In summary regarding RTC hardware, we probably don't need the Adalogger Featherwing (which has both RTC and microSD datalogging) at all!

**Comment by SUPScientist** [comment]
In order to implement robustly timed sleep, see 

https://github.com/COAST-Lab/Open-Water-Level/blob/210c09ce0191189ebe25c58f5dd1b27c464d23c0/Firmware/SLR_Boron_Maxbotix_MB7092_cm/src/SLR_Boron_Maxbotix_MB7092_cm.ino#L212-L218

and

https://github.com/COAST-Lab/Open-Water-Level/blob/210c09ce0191189ebe25c58f5dd1b27c464d23c0/Firmware/SLR_Boron_Maxbotix_MB7092_cm/src/SLR_Boron_Maxbotix_MB7092_cm.ino#L229-L238

Refer to timing constants at top of script as well.

**Comment by ebgoldstein** [comment]
This is good - I like the idea of 2 boards (and not 3). Datetime would be passed from cloud to boron to openmv (to be written on SD)

**Comment by BentleySettin** [comment]
I got the Boron to sleep for 1 minute wake up, report "Flood", and then goes back to sleep. It is not very consistent though. It will successfully cycle through 1 minute wake ups and then take 2 minutes or one time it woke at 1 minute but recorded "FloodFloodFlood".

**Comment by BentleySettin** [comment]
Here is code that tells the boron to wake up in time intervals. 

https://github.com/TinyCamML/Boron/blob/52a86aa6b2fe80acf43db42a71d94dee3eed333c/updatedBoronCode.ino#L10

https://github.com/TinyCamML/Boron/blob/52a86aa6b2fe80acf43db42a71d94dee3eed333c/updatedBoronCode.ino#L39
